usr/src/uts/common/fs/zfs/sys/spa.h

```diff
@@ -326,7 +327,8 @@ typedef enum bp_embedded_type {
 typedef struct blkptr {
 	dva_t		blk_dva[SPA_DVAS_PER_BP]; /* Data Virtual Addresses */
 	uint64_t	blk_prop;	/* size, compression, type, etc	    */
-	uint64_t	blk_pad[2];	/* Extra space for the future	    */
+	uint32_t	blk_iv[3];	/* IV for block level encryption    */
+	uint32_t	blk_pad;	/* Extra space for the future	    */
```

Comment:
Just to clarify the comment I made referencing SMB3, here's a little more detail:

> If the message should be encrypted, then we’ll skip the signing, because SMB 3 encryption ciphers (AES CCM and GCM) provide message integrity in addition to privacy.

My point was, as long as ZFS encryption uses a cipher that provides message integrity, then there should be no need to _also_ have a data "checksum" for data that's encrypted.
Hope that helps.